platform_channels/lib/src/add_pet_details.dart

```diff
@@ -16,7 +16,7 @@ class AddPetDetails extends StatefulWidget {
 
 class _AddPetDetailsState extends State<AddPetDetails> {
   final breedTextController = TextEditingController();
-  String petType = 'Dog';
+  String? petType = 'Dog';
```

Comment:
Can we make petType NNBD, and use bang operator when assigning the value of radio. This way we can make properties NNBD in the PetDetails. 

We can also use bang operator at line 33 while adding PetDetails, since even if user adds no value, the TextEditingController will return empty String.

Reply:
Sure thing @AyushBherwani1998  making petType NNBD would make sense because as petType value would never be null because it has default value Dog and either it would be Dog or Car but not null.